### PR TITLE
Auto gluon handler

### LINKED ISSUE #7889: [Integration] Integrate AutoGluon with MindsDB
The goal of this issue is to build a new handler that will integrate AutoGluon into MindsDB.

## Resources :bulb: 

:checkered_flag: To get started, please refer to the following resources:

* :books:  [Building ML Handler docs](https://docs.mindsdb.com/contribute/ml-handlers): This documentation will guide you through the process of building a custom handler for MindsDB. It contains essential information on the handler structure, methods, and best practices.

* :book:  [AutoGluon](https://auto.gluon.ai/stable/index.html)

* :point_right: [LlamaIndex Handler](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/llama_index_handler): This existing handler connects MindsDB with LlamaIndex.

## Next Steps :male_detective: :female_detective: 
 
-   You can familiarize yourself with AutoGluon.
-   Create a new handler that connects MindsDB to AutoGluon.
-   Implement the core features mentioned above.
-   Thoroughly test the integration to ensure proper functionality.
-   Submit a pull request with the new AutoGluon Integration handler.

**The PR should include the implementation of the AutoGluon handler and unit tests.**

## :loudspeaker: Additional rewards :trophy:

Each ML integration brings :seven: :zero:  : points for SWAG and entry into the draw for a  :point_down: :
- :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)
:information_source: For more info check out https://mindsdb.com/hacktoberfest/ :point_left:

### PR DESCRIPTION
**Pull request by someshfengde**
## Description
Added AutoGluon handler now we are able to train AutoGluon.TablePredictor with mindsdb 🥳 

Fixes #7889 

## Type of change

- [x] ⚡ New feature (non-breaking change which adds functionality)
- [x] 📄 This change requires a documentation update

## Additional Media:

https://github.com/mindsdb/mindsdb/assets/42097653/d09a7568-39e8-4b92-ac9e-105b6ba10d31

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by someshfengde** [comment]
> @someshfengde 
> Please add the unit tests (create a tests folder and add it there) - as mentioned in the issue description.

Oh got it can you please give example if possible llamaindex handler doesn't have  unit tests I have taken that as reference for creating this handler

**Comment by someshfengde** [comment]
let me know if you want me to move this to the handler's folder let me know

File: mindsdb/integrations/handlers/autogluon_handler/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+
+SELECT *
+FROM mindsdb.models
+WHERE name='autogluconpredictor_8';
+
+SELECT m.quality
+FROM files.red_wine_quality AS t
+JOIN mindsdb.autogluconpredictor_7 AS m;
+
```
Comment (martyna-mindsdb) [inline mindsdb/integrations/handlers/autogluon_handler/README.md:25]: Please remove the above CREATE ML_ENGINE and two SELECT statements, as these are duplicated.

**Comment by martyna-mindsdb** [comment]
@paxcema 
Please confirm the location of the tests for this handler.

File: mindsdb/integrations/handlers/autogluon_handler/README.md (hunk 1 above)
Comment (someshfengde) [reply mindsdb/integrations/handlers/autogluon_handler/README.md]: @martyna-mindsdb  removed it :)

**Comment by someshfengde** [comment]
done

**Comment by paxcema** [comment]
@martyna-mindsdb can confirm, the current directory for tests in this PR is the correct one ✅

**Review by martyna-mindsdb — APPROVED** [review]

**Comment by paxcema** [comment]
@someshfengde Moving forward, any contributions will have to pass code linting before being merged

**Pull request merged**